pkg/templates/command_groups.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templates
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// A command group is for grouping together commands
+type CommandGroup struct {
+
+	// Title for command group shown in help/usage messages
+	Header string
+
+	// List of commands for this group
+	Commands []*cobra.Command
+}
+
+type CommandGroups []CommandGroup
+
+// Add all commands from this group slice to the given command
+func (g CommandGroups) AddTo(cmd *cobra.Command) {
+	for _, group := range g {
+		for _, sub := range group.Commands {
+			cmd.AddCommand(sub)
+		}
+	}
+}
+
+// SetRootUsage sets our own help and usage function messages to the root command
+func (g CommandGroups) SetRootUsage(rootCmd *cobra.Command) {
+	engine := &templateEngine{
+		RootCmd:       rootCmd,
+		CommandGroups: g,
+	}
+	setHelpFlagsToSubCommands(rootCmd)
+	rootCmd.SilenceUsage = true
+	rootCmd.SilenceErrors = true
+	rootCmd.SetFlagErrorFunc(func(c *cobra.Command, err error) error {
+		return errors.Errorf("%s for '%s'", err.Error(), c.CommandPath())
+	})
+	rootCmd.SetUsageFunc(engine.usageFunc())
+	rootCmd.SetHelpFunc(engine.helpFunc())
+}
+
+func setHelpFlagsToSubCommands(parent *cobra.Command) {
```

Comment:
but it adds "[options]" instead of "[flags]" --> https://github.com/rhuss/knative-client/blob/41338722cdc2124dda2b3a594527749a31d8b735/pkg/templates/template_engine.go#L134

Again, inline with `kubectl` and "options" is a good wording here, as all the rest is just optional. Everything mandatory should be mentioned in the "Use" attribute.